### PR TITLE
Feat: support gitee repository in addon registry

### PR DESCRIPTION
**Pull request by fourierr**
Signed-off-by: fourierr <maxiangboo@163.com>

### Description of your changes

support gitee repository in addon registry, and the url pattern supports two valid format:
`https://gitee.com/<owner>/<repo>/tree/<branch>/<path-to-dir>`
`https://gitee.com/<owner>/<repo>/<path-to-dir>`

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/oam-dev/kubevela/blob/master/contribute/create-pull-request.md).
- [x] [Related Docs](https://github.com/oam-dev/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.

### CODE REVIEW

File: pkg/addon/addon.go [hunk 1]
```diff
@@ -418,15 +424,74 @@ func createGitHelper(content *utils.Content, token string) *gitHelper {
...
+	}
+	response, err := c.Client.Do(req.WithContext(ctx))
+	defer func() {
+		if err := response.Body.Close(); err != nil {
+			fmt.Println(err)
+		}
+	}()
+	if err != nil {
+		return nil, nil, err
+	}
```
Comment (wonderflow) [inline pkg/addon/addon.go:479]:
```suggestion
	response, err := c.Client.Do(req.WithContext(ctx))
	if err != nil {
		return nil, nil, err
	}
	defer response.Body.Close()
```

Usually we check the error first, if the err is not nil, the response can be nil. Then just defer close the body is fine, the error can be ignored and should not print out.

File: pkg/addon/addon.go [hunk 2]
```diff
@@ -418,15 +424,74 @@ func createGitHelper(content *utils.Content, token string) *gitHelper {
...
+	content, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return nil, nil, err
+	}
+	fileUnmarshalError := json.Unmarshal(content, &fileContent)
+	if fileUnmarshalError == nil {
+		return fileContent, nil, nil
```
Comment (wonderflow) [inline pkg/addon/addon.go:486]: what does it mean when the unmarshal failed?  why not return error? let's add some explanation in the code

File: pkg/addon/addon.go [hunk 3]
```diff
@@ -418,15 +424,74 @@ func createGitHelper(content *utils.Content, token string) *gitHelper {
...
+	}
+	directoryUnmarshalError := json.Unmarshal(content, &directoryContent)
+	if directoryUnmarshalError == nil {
+		return nil, directoryContent, nil
+	}
```
Comment (wonderflow) [inline pkg/addon/addon.go:491]: maybe it's better to make these lines in a single function, and write some unit tests for it.

File: pkg/addon/reader_gitee_test.go [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+// baseURLPath is a non-empty Client.BaseURL path to use during tests,
+// to ensure relative URLs are used for all endpoints. See issue #752.
+//baseURLPath = "/api-v3"
+)
+
+//var (
+//	//go:embed testdata
+//	testdata       embed.FS
+//	testdataPrefix = "testdata"
+//)
```
Comment (wonderflow) [inline pkg/addon/reader_gitee_test.go:44]: the useless code should be deleted.

File: pkg/addon/addon.go (hunk 2 above)
Comment (fourierr) [reply pkg/addon/addon.go:486]: ok, I will add the boundary error case

**Comment by zzxwill** [comment]
@wangyikewxgm Why do we still need to support Gitee as we have supported git protocol?

**Review by wonderflow — APPROVED** [review]
great job!

**Comment by wonderflow** [comment]
@zzxwill  the git protocal only cover github for now

**Review by barnettZQG — APPROVED** [review]
LGTM

**Pull request merged**